app/managers/apps_manager.ts

```diff
@@ -0,0 +1,215 @@
+
+import DatabaseManager from '@database/manager';
+import {getChannelById} from '@queries/servers/channel';
+import {getConfig, getCurrentChannelId, getCurrentTeamId, getCurrentUserId, observeConfigBooleanValue} from '@queries/servers/system';
+import {validateBindings} from '@utils/apps';
+import {logDebug} from '@utils/log';
+
+import NetworkManager from './network_manager';
+
+const emptyBindings: AppBinding[] = [];
+
+class AppsManager {
+    private enabled: {[serverUrl: string]: BehaviorSubject<boolean>} = {};
+
+    private bindings: {[serverUrl: string]: BehaviorSubject<AppBinding[]>} = {};
+    private threadBindings: {[serverUrl: string]: BehaviorSubject<{channelId: string; bindings: AppBinding[]}>} = {};
+
+    private commandForms: {[serverUrl: string]: {[location: string]: AppForm}} = {};
+    private threadCommandForms: {[serverUrl: string]: {[location: string]: AppForm}} = {};
+
+    private getEnabledSubject = (serverUrl: string) => {
+        if (!this.enabled[serverUrl]) {
+            this.enabled[serverUrl] = new BehaviorSubject(true);
+        }
+
+        return this.enabled[serverUrl];
+    };
+
+    private getBindingsSubject = (serverUrl: string) => {
+        if (!this.bindings[serverUrl]) {
+            this.bindings[serverUrl] = new BehaviorSubject([]);
+        }
+
+        return this.bindings[serverUrl];
+    };
+
+    private getThreadsBindingsSubject = (serverUrl: string) => {
+        if (!this.threadBindings[serverUrl]) {
+            this.threadBindings[serverUrl] = new BehaviorSubject({channelId: '', bindings: emptyBindings});
+        }
+
+        return this.threadBindings[serverUrl];
+    };
+
+    private handleError = (serverUrl: string) => {
+        const enabled = this.getEnabledSubject(serverUrl);
+        if (enabled.value) {
+            enabled.next(false);
+        }
+        this.getBindingsSubject(serverUrl).next(emptyBindings);
+        this.getThreadsBindingsSubject(serverUrl).next({channelId: '', bindings: emptyBindings});
+
+        this.commandForms[serverUrl] = {};
+        this.threadCommandForms[serverUrl] = {};
+    };
+
+    removeServer = (serverUrl: string) => {
+        delete (this.enabled[serverUrl]);
+
+        delete (this.bindings[serverUrl]);
+        delete (this.threadBindings[serverUrl]);
+
+        delete (this.commandForms[serverUrl]);
+        delete (this.threadCommandForms[serverUrl]);
+    };
+
+    clearServer = (serverUrl: string) => {
+        this.clearBindings(serverUrl);
+        this.clearBindings(serverUrl, true);
+        this.commandForms[serverUrl] = {};
+        this.threadCommandForms[serverUrl] = {};
+    };
+
+    isAppsEnabled = async (serverUrl: string) => {
+        try {
+            const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+            const config = await getConfig(database);
+            return this.getEnabledSubject(serverUrl).value && config?.FeatureFlagAppsEnabled === 'true';
```

Comment:
The `AppsEnabled` flag is enabled by default now, so it's only relevant on older servers

To determine if we should be interacting with the App framework, we should keep track of the Apps plugin's presence as it is in mobile v1 https://github.com/mattermost/mattermost-mobile/pull/5673:

https://github.com/mattermost/mattermost-mobile/blob/ccdde858c7402620d2003d2b0757ea51d15682c5/app/utils/apps.ts#L9-L20

Possibly out of scope of this PR

Reply:
The "enabledSubject" would be the "appsPluginIsEnabled" in the code shared. I think I catch the different cases of the reducers in V1, but feel free to prove me wrong :smile:

If you see any problem, let me know and I will fix it.

Reply:
The "enabledSubject" would be the "appsPluginIsEnabled" in the code shared. I think I catch the different cases of the reducers in V1, but feel free to prove me wrong :smile: 

If you see any problem, let me know and I will fix it.